ocp_resources/node_network_configuration_policy.py

```diff
@@ -374,21 +374,28 @@ def wait_for_status_success(self):
         failed_condition_reason = self.Conditions.Reason.FAILED_TO_CONFIGURE
         no_match_node_condition_reason = self.Conditions.Reason.NO_MATCHING_NODE
 
-        # if we get here too fast there are no conditions, we need to wait.
-        self.wait_for_configuration_conditions_unknown_or_progressing()
-
-        samples = TimeoutSampler(wait_timeout=self.success_timeout, sleep=1, func=lambda: self.status)
         try:
-            for sample in samples:
-                if sample == self.Conditions.Reason.SUCCESSFULLY_CONFIGURED:
-                    self.logger.info(f"NNCP {self.name} configured Successfully")
-                    return sample
-
-                elif sample == no_match_node_condition_reason:
-                    raise NNCPConfigurationFailed(f"{self.name}. Reason: {no_match_node_condition_reason}")
-
-                elif sample == failed_condition_reason:
-                    self._process_failed_status(failed_condition_reason=failed_condition_reason)
+            for sample in TimeoutSampler(
+                wait_timeout=self.success_timeout,
+                sleep=5,
+                func=lambda: next(
```

Comment:
Why did you change it? It was more clean and readable before.

Reply:
Based on https://github.com/RedHatQE/openshift-python-wrapper/pull/2289#discussion_r1927430258 I changed it. Please let me know what you want me to do next.

Reply:
@myakove , I thought you care about efficiency. This is more efficient than re-defining the function on each call, and using generators to reduce memory consumption. And regarding readability, it’s opinionated, for me this is shorter and clearer.
However, this is your call, none of the reasons above merit blocking the PR.

Reply:
ack